src/Messages.jl

```diff
@@ -455,25 +459,20 @@ function decode(m::Message, encoding::String="gzip")::Vector{UInt8}
 end
 
 # Writing HTTP Messages to IO streams
-"""
-    HTTP.httpversion(::Message)
-
-e.g. `"HTTP/1.1"`
-"""
-httpversion(m::Message) = "HTTP/$(m.version.major).$(m.version.minor)"
+Base.write(io::IO, v::HTTPVersion) = write(io, "HTTP/", string(v.major), ".", string(v.minor))
 
 """
     writestartline(::IO, ::Message)
 
 e.g. `"GET /path HTTP/1.1\\r\\n"` or `"HTTP/1.1 200 OK\\r\\n"`
 """
 function writestartline(io::IO, r::Request)
-    write(io, "$(r.method) $(r.target) $(httpversion(r))\r\n")
+    write(io, r.method, " ", r.target, " ", r.version, "\r\n")
```

Comment:
So instead of creating a big string and writing that, we iterate over the args and each is written to io?

Reply:
yeah, exactly, we save on allocating that big string before writing it out
https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-string-interpolation-for-I/O

and we overload `Base.write` for `version` so it also writes what we want straight to IO

so each saving 1 string allocation